torch/utils/data/dataloader.py

```diff
@@ -521,12 +522,36 @@ def _create_warning_msg(num_worker_suggest, num_worker_created, cpuset_checked):
                 self.num_workers,
                 cpuset_checked))
 
+    def _get_shared_seed(self):
+        if isinstance(self.dataset, IterDataPipe):
+            _shared_tensor_seed = torch.empty((), dtype=torch.int64).random_(generator=self.generator)
+            if dist.is_available() and dist.is_initialized():
+                rank = dist.get_rank()
+                if rank == 0:
+                    ws = dist.get_world_size()
+                    reqs = []
+                    for rank_idx in range(1, ws):
+                        req = dist.isend(tensor=_shared_tensor_seed, dst=rank_idx)
```

Comment:
Feels like you need to `tag = 'dataloader_shared_seed'` here

Reply:
It totally makes sense. Since `tag` only accepts integer, I will use `rank` as the tag to transmit `_shared_tensor_seed`

Reply:
It looks like an unused feature inside of the `dist` code. As for me `tag` should prevent other parts of the code from receiving this tensor. `rank_id` is fine for now, but It is better to talk with dist team, to figure out if we need to define tag constants somewhere and have something like `_DIST_TAG_DOWNLOADER_ = 1000`

Reply:
Reasonable to me!